File: src/main/java/edu/kit/ifv/mobitopp/actitopp/ChoiceFunction.java
package edu.kit.ifv.mobitopp.actitopp;

import java.util.List;

public interface ChoiceFunction
{

    /**
     * Calculates the probabilitites for the alternatives
     * @param alternatives
     * @return
     */
    public void calculateProbabilities(List<DCAlternative> alternatives);
    
    /**
     * Returns the index of the choice alternative that has been chosen
     * @param alternatives
     * @param random
     * @return
     */
    public int chooseAlternative(List<DCAlternative> alternatives, double random);
}

File: src/main/java/edu/kit/ifv/mobitopp/actitopp/ChoiceFunction.kt
package edu.kit.ifv.mobitopp.actitopp

interface ChoiceFunction {
    /**
     * Calculates the probabilitites for the alternatives
     *
     * @param alternatives
     */
    fun calculateProbabilities(alternatives: List<DCAlternative>)

    /**
     * Returns the index of the choice alternative that has been chosen
     *
     * @param alternatives
     * @param random
     * @return
     */
    fun chooseAlternative(alternatives: List<DCAlternative>, random: Double): Int
}
